chemprop/v2/cli/train.py

```diff
@@ -239,22 +239,21 @@ def add_train_args(parser: ArgumentParser) -> ArgumentParser:
     # data_args is added in add_common_args()
     data_args.add_argument(
         "--target-columns",
-        type=list,
+        nargs="+",
         help="Name of the columns containing target values. By default, uses all columns except the SMILES column and the :code:`ignore_columns`.",
     )
     data_args.add_argument(
         "--ignore-columns",
-        type=list,
+        nargs="+",
         help="Name of the columns to ignore when :code:`target_columns` is not provided.",
     )
 
     data_args.add_argument(
         "-t",
         "--task-type",
         default="regression",
-        action=RegistryAction(
-            ReadoutRegistry
-        ),  # TODO: is this correct? The choices should be ['regression', 'classification', 'multiclass', 'spectra']
+        choices=ReadoutRegistry.keys(),  # TODO: is this correct? The choices should be ['regression', 'classification', 'multiclass', 'spectra']
```

Comment:
```suggestion
```
the `RegistryAction` class factory automatically sets `choices` to be all the keys of the respective registry

Reply:
@davidegraff Thanks! Yeah, you're right. I reverted this change and tested, and `--task-type` worked. I dropped this commit in the new push.